include/seqan3/alphabet/nucleotide/nucl16.hpp

```diff
@@ -120,79 +103,109 @@ struct nucl16
-    constexpr char_type to_char() const
+    //!\brief Return the letter's numeric value or rank in the alphabet.
+    constexpr rank_type to_rank() const noexcept
     {
-        return value_to_char[static_cast<integral_type>(value)];
+        return static_cast<rank_type>(_value);
     }
+    //!\}
 
-    //! assign from a character
-    constexpr nucl16 from_char(char_type const c)
+    /*!\name Write functions
+     * \{
+     */
+    //!\brief Assign from a character.
+    constexpr nucl16 & assign_char(char_type const c) noexcept
     {
-        value = char_to_value[c];
+        _value = char_to_value[c];
         return *this;
     }
 
-    //! return the letter's numeric value or rank in the alphabet
-    constexpr integral_type to_integral() const
-    {
-        return static_cast<integral_type>(value);
-    }
-
-    //! assign from a numeric value
-    constexpr nucl16 from_integral(integral_type const c)
+    //!\brief Assign from a numeric value.
+    constexpr nucl16 & assign_rank(rank_type const c)
     {
         assert(c < value_size);
-        value = static_cast<internal_type>(c);
+        _value = static_cast<internal_type>(c);
         return *this;
     }
+    //!\}
 
-    //! The size of the alphabet, i.e. the number of different values it can take.
-    static constexpr integral_type value_size{16};
+    //!\brief The size of the alphabet, i.e. the number of different values it can take.
+    static constexpr rank_type value_size{16};
 
-    //! @name comparison operators
-    //!@{
-    constexpr bool operator==(nucl16 const & rhs) const
+    //!\name Comparison operators
+    //!\{
+    constexpr bool operator==(nucl16 const & rhs) const noexcept
     {
-        return value == rhs.value;
+        return _value == rhs._value;
     }
 
-    constexpr bool operator!=(nucl16 const & rhs) const
+    constexpr bool operator!=(nucl16 const & rhs) const noexcept
     {
-        return value != rhs.value;
+        return _value != rhs._value;
     }
 
-    constexpr bool operator<(nucl16 const & rhs) const
+    constexpr bool operator<(nucl16 const & rhs) const noexcept
     {
-        return value < rhs.value;
+        return _value < rhs._value;
     }
 
-    constexpr bool operator>(nucl16 const & rhs) const
+    constexpr bool operator>(nucl16 const & rhs) const noexcept
     {
-        return value > rhs.value;
+        return _value > rhs._value;
     }
 
-    constexpr bool operator<=(nucl16 const & rhs) const
+    constexpr bool operator<=(nucl16 const & rhs) const noexcept
     {
-        return value <= rhs.value;
+        return _value <= rhs._value;
     }
 
-    constexpr bool operator>=(nucl16 const & rhs) const
+    constexpr bool operator>=(nucl16 const & rhs) const noexcept
     {
-        return value >= rhs.value;
+        return _value >= rhs._value;
     }
     //!@}
 
-    //! \privatesection
-    // conversion tables
+protected:
+    //!\privatesection
```

Comment:
it is not necessary, because there is nothing following it than shall become public again (also the public keyword implicitly begins a public section again).